src/Processors/Executors/PipelineExecutor.h

```diff
@@ -85,6 +85,7 @@ class PipelineExecutor
     AcquiredSlotPtr single_thread_cpu_slot; // cpu slot for single-thread mode to work using executeStep()
     std::unique_ptr<ThreadPool> pool;
     std::atomic_size_t threads = 0;
+    std::mutex spawn_lock;
```

Comment:
Please use TSA annotations to document what the lock protects (... grep "TSA_" in the source code for examples).